### PR TITLE
feat: support webpack config optimize

### PR DESCRIPTION
**Pull request by justquanyin**
## Checklist
* [x] `npm test` passes
* [x] tests are included
* [x] documentation is changed or added
* [x] commit message follows commit guidelines

## Description of change

修改webpack配置，支持更方便的自定义式的配置改写。

### CODE REVIEW

File: packages/beidou-webpack/README.md [hunk 1]
```diff
@@ -99,9 +99,308 @@ module.exports = (app, defaultConfig, dev, target) => {
     //something else to override
   };
 };
+
+// Exapple 2:
```
Comment (njugray) [inline packages/beidou-webpack/README.md]:
```suggestion
// Example 2:
```

File: packages/beidou-webpack/lib/builder/index.js [hunk 2]
```diff
@@ -1,13 +1,12 @@
 'use strict';
 
-const webpack = require('webpack');
 const helper = require('../utils');
+const webpack = require('webpack');
```
Comment (njugray) [inline packages/beidou-webpack/lib/builder/index.js]: ??

Comment (justquanyin) [reply packages/beidou-webpack/lib/builder/index.js]: 之前改过一次自定义webpack，所以把webpack依赖去掉了；后面没有再支持自定义webpack，所以依赖包就有重新引入了。导致删了上面，加在了下面

File: packages/beidou-webpack/lib/factory/Plugin.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+const is = require('is');
+
+class Plugin {
+  constructor(model, options, alias) {
+    if (is.function(model)) {
+      this.model = model;
+      this.options = options;
+      this.alias = alias || model.name;
+    } else if (is.object(model)) {
+      this.class = model;
```
[the remark this answers is not in the document]
Comment (justquanyin) [reply packages/beidou-webpack/lib/factory/Plugin.js]: 已修改

File: packages/beidou-webpack/README.md [hunk 4]
```diff
@@ -99,6 +99,97 @@ module.exports = (app, defaultConfig, dev, target) => {
...
+  factory.set('output',{
+    {
+      path: outputPath,
+      filename: '[name].js?[hash]',
+      chunkFilename: '[name].modify.js',
+      publicPath: '/build/',
+    }
+  })
+  // modify default webpack config
+  factory.modifyPlugin('CommonsChunkPlugin',
```
[the remark this answers is not in the document]
Comment (justquanyin) [reply packages/beidou-webpack/README.md]: 已修改

**Pull request merged**